packages/lexical-selection/src/range-selection.ts

```diff
@@ -94,6 +94,11 @@ export function $setBlocksType<T extends ElementNode>(
   for (const node of selection.getNodes()) {
     if ($isElementNode(node) && INTERNAL_$isBlock(node)) {
       blockMap.set(node.getKey(), node);
+    } else {
```

Comment:
```suggestion
    } else if (anchorAndFocus === null) {
```

This shouldn't be necessary for RangeSelection due to the anchorAndFocus code above (all containing blocks are either fully inside the RangeSelection or ancestors of the anchor or focus)